src/main/java-templates/org/primefaces/component/slider/SliderTemplate.java

```diff
@@ -57,3 +60,60 @@ public void queueEvent(FacesEvent event) {
             super.queueEvent(event);
         }
     }
+
+    public final static String VALUE_OUT_OF_RANGE = "primefaces.slider.OUT_OF_RANGE";
+
+    @Override
+    public void validate(FacesContext context) {
+        super.validate(context);
+        
+        if(isValid()) {
+            String[] inputIds = this.getFor().split(",");
+            if (this.isRange()) {
+                UIInput inputFrom = (UIInput) SearchExpressionFacade.resolveComponent(context, this, inputIds[0].trim());
+                UIInput inputTo = (UIInput) SearchExpressionFacade.resolveComponent(context, this, inputIds[1].trim());
+                String valueFromStr = getSubmittedValue(inputFrom).toString();
+                String valueToStr = getSubmittedValue(inputTo).toString();
+                double valueFrom = Double.valueOf(valueFromStr);
+                double valueTo = Double.valueOf(valueToStr);
+                if (valueFrom < this.getMinValue() || valueFrom > this.getMaxValue()) {
+                    this.setValid(false);
```

Comment:
Why not use `non-shortcircuit or` to force evaluating both statements ?

```
if(valueFrom < this.getMinValue() | valueFrom > this.getMaxValue())
  setInvaliad
```

Reply:
Nevermind, my bad. Your code is OK. You are evaluating two different fields.

Reply:
Just a nitpick

if valueFrom and valueTo are both out of range, there is no need to excute `valueTo < valueFrom` since both field are already invalid at that point. 

Those three if statements will execute in order regardless of the condition is valid or invalid that can waste some cycles.

Reply:
Let's say `valueFrom=2, valueTo=1, minValue=0, maxValue=2`. 

Then the first two conditions aren't met, however the third condition is since `valueTo < valueFrom`. So the third condition is absolutely required.

Your suggestion would be to additionally check these prerequisites in the third condition? Then you would save the setter but introduce getter cycles. Does that make sense?

Reply:
> Your suggestion would be to additionally check these prerequisites in the third condition? Then you would save the setter but introduce getter cycles. Does that make sense?

No, what I meant was to move the third condition to the first.
```
if (valueTo < valueFrom) {  
    // both are invalid then no need to check individual fields
} else {
 //  checkIndividual fields
  if (valueFrom) {

  }

  if (valueTo) {

  }
}
```

instead of 

```
if (valueFrom) {

}

if (valueTo) {

}

if (valueFrom > valueTo) {

}
```
---

Let's say `valueFrom = 4, valueTo = 0, minValue = 1, maxValue = 3`

 `if (valueFrom < this.getMinValue() || valueFrom > this.getMaxValue())` will be true

`if (valueTo > this.getMaxValue() || valueTo < this.getMinValue())` will be true

`if (valueTo < valueFrom)` will also be true (wasted)

---

Let's say valueFrom=2, valueTo=1, minValue=0, maxValue=2.

 `if (valueFrom < this.getMinValue() || valueFrom > this.getMaxValue())` will be false

`if (valueTo > this.getMaxValue() || valueTo < this.getMinValue())` will be false

`if (valueTo < valueFrom)` will also be true

the third case is necessary but it will also be faster if we check valueTo < valueFrom first.